src/views/Pools/components/PoolsTable/ActionPanel/ActionPanel.tsx

```diff
@@ -0,0 +1,241 @@
+import Balance from 'components/Balance'
+import { CompoundingPoolTag, ManualPoolTag } from 'components/Tags'
+import { getAddress } from 'utils/addressHelpers'
+import { registerToken } from 'utils/wallet'
+import { getBalanceNumber } from 'utils/formatBalance'
+import Harvest from './Harvest'
+import Stake from './Stake'
+import Apr from '../Apr'
+
+const expandAnimation = keyframes`
+  from {
+    max-height: 0px;
+  }
+  to {
+    max-height: 500px;
+  }
+`
+
+const collapseAnimation = keyframes`
+  from {
+    max-height: 500px;
+  }
+  to {
+    max-height: 0px;
+  }
+`
+
+const StyledActionPanel = styled.div<{ expanded: boolean }>`
+  animation: ${({ expanded }) =>
+    expanded
+      ? css`
+          ${expandAnimation} 300ms linear forwards
+        `
+      : css`
+          ${collapseAnimation} 300ms linear forwards
+        `};
+  overflow: hidden;
+  background: ${({ theme }) => theme.colors.dropdown};
+  display: flex;
+  flex-direction: column-reverse;
+  justify-content: center;
+  padding: 24px;
+
+  ${({ theme }) => theme.mediaQueries.lg} {
+    flex-direction: row;
+    padding: 16px 32px;
+  }
+`
+
+const ActionContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+
+  ${({ theme }) => theme.mediaQueries.sm} {
+    flex-direction: row;
+    align-items: center;
+    flex-grow: 1;
+    flex-basis: 0;
+  }
+`
+
+type MediaBreakpoints = {
+  isXs: boolean
+  isSm: boolean
+  isMd: boolean
+  isLg: boolean
+  isXl: boolean
+}
+
+interface ActionPanelProps {
+  account: string
+  pool: Pool
+  isAutoVault: boolean
+  userDataLoaded: boolean
+  expanded: boolean
+  breakpoints: MediaBreakpoints
+}
+
+const InfoSection = styled(Box)`
+  flex: 0 0 210px;
+  padding: 8px 8px;
+  ${({ theme }) => theme.mediaQueries.lg} {
+    padding: 0;
+  }
+`
+
+const ActionPanel: React.FC<ActionPanelProps> = ({
+  account,
+  pool,
+  isAutoVault,
+  userDataLoaded,
+  expanded,
+  breakpoints,
+}) => {
+  const { sousId, stakingToken, earningToken, startBlock, totalStaked, endBlock, isFinished } = pool
+  const { t } = useTranslation()
+  const { currentBlock } = useBlock()
+  const { isXs, isSm, isMd } = breakpoints
+
+  const shouldShowBlockCountdown = Boolean(!isFinished && startBlock && endBlock)
+  const blocksUntilStart = Math.max(startBlock - currentBlock, 0)
+  const blocksRemaining = Math.max(endBlock - currentBlock, 0)
+  const hasPoolStarted = blocksUntilStart === 0 && blocksRemaining > 0
+  const blocksToDisplay = hasPoolStarted ? blocksRemaining : blocksUntilStart
+
+  const isMetaMaskInScope = !!(window as WindowChain).ethereum?.isMetaMask
+  const tokenAddress = earningToken.address ? getAddress(earningToken.address) : ''
+  const imageSrc = `${BASE_URL}/images/tokens/${earningToken.symbol.toLowerCase()}.png`
+
+  const {
+    totalCakeInVault,
+    fees: { performanceFee },
+  } = useCakeVault()
+
+  const performanceFeeAsDecimal = performanceFee && performanceFee / 100
+  const isManualCakePool = sousId === 0
```

Comment:
Is this sort of thing (not all of it, but a lot of it) the kind of thing you were talking about in terms of having it in helpers, as it's used in a few places?